dtfs-central-api/src/services/tfm/deal-cancellation.service.ts

```diff
@@ -1,16 +1,38 @@
-import { ACTIVITY_TYPES, InvalidAuditDetailsError, TfmActivity, TfmAuditDetails, TfmDealCancellation, TfmDealCancellationResponse } from '@ukef/dtfs2-common';
+import {
+  ACTIVITY_TYPES,
+  AuditDetails,
+  DEAL_STATUS,
+  InvalidAuditDetailsError,
+  TfmActivity,
+  TfmAuditDetails,
+  TfmDealCancellation,
+  TfmDealCancellationResponse,
+} from '@ukef/dtfs2-common';
 import { ObjectId } from 'mongodb';
-import { getUnixTime } from 'date-fns';
+import { endOfDay, getUnixTime, isAfter, toDate } from 'date-fns';
 import { TfmDealCancellationRepo } from '../../repositories/tfm-deals-repo';
 import { TfmUsersRepo } from '../../repositories/tfm-users-repo';
+import { PortalDealService } from '../portal/deal.service';
 
 export class DealCancellationService {
+  /**
+   * Submit a deal cancellation, either cancelling the deal or scheduling it for the
+   * future depending on the effective date
+   *
+   * @param dealId The deal id to be cancelled
+   * @param cancellation - the cancellation
+   * @param auditDetails - the users audit details
+   * @returns Tfm deal cancellation response
+   */
   public static async cancelDeal(
     dealId: ObjectId | string,
     cancellation: TfmDealCancellation,
     auditDetails: TfmAuditDetails,
-  ): Promise<TfmDealCancellationResponse | undefined> {
-    const isDealCancellationPastOrPresent = new Date().valueOf() >= cancellation.effectiveFrom;
+  ): Promise<TfmDealCancellationResponse> {
+    const effectiveFromDate = toDate(cancellation.effectiveFrom);
+    const endOfToday = endOfDay(new Date());
+
+    const isDealCancellationPastOrPresent = !isAfter(effectiveFromDate, endOfToday);
```

Comment:
I'd probably look at just having this as `isDealCancellationInFuture` but no need to change, just my opinion